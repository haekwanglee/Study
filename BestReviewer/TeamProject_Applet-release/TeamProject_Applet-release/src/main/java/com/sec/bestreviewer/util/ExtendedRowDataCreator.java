package com.sec.bestreviewer.util;

import com.sec.bestreviewer.database.Row;
import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;
import java.util.LinkedList;
import java.util.List;

public class ExtendedRowDataCreator {
    private static final String NAME_DELIMITER = " ";
    private static final String PHONE_NUM_DELIMITER = "-";
    private static final String DATE_PATTERN = "yyyyMMdd";

    private static final int FIRST_NAME_INDEX = 0;
    private static final int LAST_NAME_INDEX = 1;

    private static final int MIDDLE_NUMBER_INDEX = 1;
    private static final int LAST_NUMBER_INDEX = 2;

    public static Row create(Row row) {
        final List<String> updatedDataList = new LinkedList<>(row.getRow());

        appendNames(row, updatedDataList);
        appendPhoneNumbers(row, updatedDataList);
        appendDates(row, updatedDataList);

        return new Row(updatedDataList);
    }

    private static void appendNames(Row row, List<String> extendedList) {
        final String name = row.getValue(EmployeeSchema.NAME.getIndex());
        final String[] splitName = name.split(NAME_DELIMITER);
        extendedList.add(splitName[FIRST_NAME_INDEX]);
        extendedList.add(splitName[LAST_NAME_INDEX]);
    }

    private static void appendPhoneNumbers(Row row, List<String> updatedDataList) {
        final String phoneNumber = row.getValue(EmployeeSchema.PHONE_NUMBER.getIndex());
        final String[] splitPhoneNumber = phoneNumber.split(PHONE_NUM_DELIMITER);
        updatedDataList.add(splitPhoneNumber[MIDDLE_NUMBER_INDEX]);
        updatedDataList.add(splitPhoneNumber[LAST_NUMBER_INDEX]);
    }

    private static void appendDates(Row row, List<String> updatedDataList) {
        final String birthDay = row.getValue(EmployeeSchema.BIRTHDAY.getIndex());
        LocalDate date = LocalDate.parse(birthDay, DateTimeFormatter.ofPattern(DATE_PATTERN));
        updatedDataList.add(String.valueOf(date.getYear()));
        updatedDataList.add(getDateString(date.getMonthValue()));
        updatedDataList.add(getDateString(date.getDayOfMonth()));
    }

    private static String getDateString(int monthOrDay) {
        String date = String.valueOf(monthOrDay);
        return date.length() == 1
                ? "0" + date
                : date;
    }
}
