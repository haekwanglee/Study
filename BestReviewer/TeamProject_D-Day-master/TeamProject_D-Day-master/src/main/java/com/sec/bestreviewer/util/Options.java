package com.sec.bestreviewer.util;

public class Options {
    public static final String LAST_NAME_OPTION = "-l";
    public static final String MIDDLE_NUMBER_OPTION = "-m";
    public static final String LAST_NUMBER_OPTION = "-l";
    public static final String YEAR_OPTION = "-y";
    public static final String MONTH_OPTION = "-m";
    public static final String DAY_OPTION = "-d";
    public static final String GRATER_OPTION = "-g";
    public static final String GRATER_EQUAL_OPTION = "-ge";
    public static final String SMALLER_OPTION = "-s";
    public static final String SMALLER_EQUAL_OPTION = "-se";
    public static final String FIRST_NAME_OPTION = "-f";
    public static final String EMPTY_OPTION = " ";
}
