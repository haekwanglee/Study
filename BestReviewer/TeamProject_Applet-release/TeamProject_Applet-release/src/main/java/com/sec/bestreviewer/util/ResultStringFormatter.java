package com.sec.bestreviewer.util;

import com.sec.bestreviewer.database.Row;
import com.sec.bestreviewer.database.field.EmployeeSchema;

import java.util.Comparator;
import java.util.List;
import java.util.stream.Collectors;

public class ResultStringFormatter {
    private static final int YEAR_1990_EMPLOYEE_NUMBER = 90_000000;
    private static final int YEAR_PREFIX_19 = 1900_000000;
    private static final int YEAR_PREFIX_20 = 2000_000000;

    public static List<String> getEmployeeListToFormattedString(List<Row> employeeList) {
        return employeeList.stream()
                .sorted(Comparator.comparing(ResultStringFormatter::getEmployeeNumber))
                .map(ResultStringFormatter::getEmployeeToFormattedString)
                .collect(Collectors.toList());
    }

    private static Integer getEmployeeNumber(Row employee) {
        int employeeNumber = Integer.parseInt(employee.getValue(EmployeeSchema.getPrimaryKeyIndex()));
        return employeeNumber >= YEAR_1990_EMPLOYEE_NUMBER ?
                YEAR_PREFIX_19 + employeeNumber : YEAR_PREFIX_20 + employeeNumber;
    }

    public static String getEmployeeToFormattedString(Row employee) {
        return employee.getValue(EmployeeSchema.EMPLOYEE_NUMBER.getIndex()) + "," +
                employee.getValue(EmployeeSchema.NAME.getIndex()) + "," +
                employee.getValue(EmployeeSchema.CAREER_LEVEL.getIndex()) + "," +
                employee.getValue(EmployeeSchema.PHONE_NUMBER.getIndex()) + "," +
                employee.getValue(EmployeeSchema.BIRTHDAY.getIndex()) + "," +
                employee.getValue(EmployeeSchema.CERTI.getIndex());

    }
}