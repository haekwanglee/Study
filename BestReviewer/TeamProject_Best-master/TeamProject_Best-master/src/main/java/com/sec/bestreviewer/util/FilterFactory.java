package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.filter.AndFilter;
import com.sec.bestreviewer.store.filter.Filter;
import com.sec.bestreviewer.store.filter.OrFilter;
import com.sec.bestreviewer.store.filter.SingleFilter;

import java.util.List;

import static com.sec.bestreviewer.util.ParamsParser.PairIndex.FIRST_PAIR;
import static com.sec.bestreviewer.util.ParamsParser.PairIndex.SECOND_PAIR;

public class FilterFactory {

    public static Filter buildFilter(List<String> params, OptionParser optionParser) {
        if (optionParser.getAndOption().isAndOn()) {
            Pair<String, String> conditionPair1 = ParamsParser.parser(params, FIRST_PAIR);
            OptionParser optionParser1 = new OptionParser(optionParser.getAndOption().getFirstOptions(), conditionPair1.fieldName);

            Pair<String, String> conditionPair2 = ParamsParser.parser(params, SECOND_PAIR);
            OptionParser optionParser2 = new OptionParser(optionParser.getAndOption().getSecondOptions(), conditionPair2.fieldName);
            return new AndFilter(new SingleFilter(conditionPair1, optionParser1), new SingleFilter(conditionPair2, optionParser2));
        }

        if (optionParser.getOrOption().isOrOn()) {
            Pair<String, String> conditionPair1 = ParamsParser.parser(params, FIRST_PAIR);
            OptionParser optionParser1 = new OptionParser(optionParser.getOrOption().getFirstOptions(), conditionPair1.fieldName);

            Pair<String, String> conditionPair2 = ParamsParser.parser(params, SECOND_PAIR);
            OptionParser optionParser2 = new OptionParser(optionParser.getOrOption().getSecondOptions(), conditionPair2.fieldName);

            return new OrFilter(new SingleFilter(conditionPair1, optionParser1), new SingleFilter(conditionPair2, optionParser2));
        }

        Pair<String, String> conditionPair = ParamsParser.parser(params, FIRST_PAIR);
        return new SingleFilter(conditionPair, optionParser);
    }
}
