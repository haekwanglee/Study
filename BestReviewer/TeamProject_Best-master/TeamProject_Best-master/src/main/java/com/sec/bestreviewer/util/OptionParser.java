package com.sec.bestreviewer.util;

import com.sec.bestreviewer.option.*;

import java.util.Arrays;
import java.util.List;

public class OptionParser {

    public final static List<String> SUPPORTED_OPTIONS = Arrays.asList("-p", "-f", "-l", "-m", "-y", "-d", "-g", "-ge", "-s", "-se", "-a", "-o");
    private final PrintOption printOption;
    private final NameOption nameOption;
    private final BirthDayOption birthDayOption;
    private final PhoneOption phoneOption;
    private final EqualityOption equalityOption;
    private final AndOption andOption;
    private final OrOption orOption;

    public OptionParser(List<String> options, String fieldName) {
        printOption = new PrintOption(options);
        nameOption = new NameOption(options, fieldName);
        birthDayOption = new BirthDayOption(options, fieldName);
        phoneOption = new PhoneOption(options, fieldName);
        equalityOption = new EqualityOption(options);
        andOption = new AndOption(options);
        orOption = new OrOption(options);
    }

    public PrintOption getPrintOption() {
        return printOption;
    }

    public NameOption getNameOption() {
        return nameOption;
    }

    public BirthDayOption getBirthDayOption() {
        return birthDayOption;
    }

    public PhoneOption getPhoneOption() {
        return phoneOption;
    }

    public EqualityOption getEqualityOption() {
        return equalityOption;
    }

    public AndOption getAndOption() {
        return andOption;
    }

    public OrOption getOrOption() {
        return orOption;
    }
}
