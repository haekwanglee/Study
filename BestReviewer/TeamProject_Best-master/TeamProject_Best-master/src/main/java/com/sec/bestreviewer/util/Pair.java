package com.sec.bestreviewer.util;

import java.util.Objects;

public class Pair<F, S> {
    public final F fieldName;
    public final S value;

    private Pair(F fieldName, S value) {
        this.fieldName = fieldName;
        this.value = value;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Pair)) {
            return false;
        }
        Pair<?, ?> p = (Pair<?, ?>) o;
        return Objects.equals(p.fieldName, fieldName) && Objects.equals(p.value, value);
    }

    @Override
    public int hashCode() {
        return (fieldName == null ? 0 : fieldName.hashCode()) ^ (value == null ? 0 : value.hashCode());
    }

    @Override
    public String toString() {
        return "Pair{" + String.valueOf(fieldName) + " " + String.valueOf(value) + "}";
    }

    public static <A, B> Pair <A, B> create(A a, B b) {
        return new Pair<A, B>(a, b);
    }
}