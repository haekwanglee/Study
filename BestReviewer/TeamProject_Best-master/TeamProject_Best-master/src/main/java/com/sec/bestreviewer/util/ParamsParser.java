package com.sec.bestreviewer.util;

import com.sec.bestreviewer.store.EmployeeStore;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ParamsParser {

    public enum PairIndex {
        FIRST_PAIR(0), SECOND_PAIR(2), THIRD_PAIR(4);

        private final int index;
        PairIndex(int index) {
            this.index = index;
        }
    }

    private static final Map<String, String> fieldMap = new HashMap<>();
    private static final String EMPLOYEE_NUMBER = "employeeNum";
    private static final String NAME = "name";
    private static final String CAREER_LEVEL = "cl";
    private static final String PHONE_NUMBER = "phoneNum";
    private static final String BIRTHDAY = "birthday";
    private static final String CERTI = "certi";

    // mapping table between commands name and DB columns name
    static {
        fieldMap.put(EMPLOYEE_NUMBER, EmployeeStore.FIELD_EMPLOYEE_NUMBER);
        fieldMap.put(NAME, EmployeeStore.FIELD_NAME);
        fieldMap.put(CAREER_LEVEL, EmployeeStore.FIELD_CAREER_LEVEL);
        fieldMap.put(PHONE_NUMBER, EmployeeStore.FIELD_PHONE_NUMBER);
        fieldMap.put(BIRTHDAY, EmployeeStore.FIELD_BIRTH_DAY);
        fieldMap.put(CERTI, EmployeeStore.FIELD_CERTI);
    }

    public static Pair<String, String> parser(List<String> params, PairIndex pairIndex) {
        return getMapFromParams(params, pairIndex.index);
    }

    private static Pair<String, String> getMapFromParams(List<String> params, int index) {
        try {
            final String fieldName = fieldMap.get(params.get(index));
            if (fieldName == null) {
                throw new IllegalArgumentException("Wrong field: " + params.get(index));
            }
        } catch (ArrayIndexOutOfBoundsException ex) {
            throw new IllegalArgumentException("Wrong parameters");
        }
        return Pair.create(fieldMap.get(params.get(index)), params.get(index + 1));
    }
}
