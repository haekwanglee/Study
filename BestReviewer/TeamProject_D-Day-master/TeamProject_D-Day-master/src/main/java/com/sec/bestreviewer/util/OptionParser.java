package com.sec.bestreviewer.util;

import java.util.List;

public class OptionParser {

    private static final int NUMBER_OF_OPTION_SIZE_FOR_FIRST_COMMAND = 3;   //first command consists of '-p', '-f/l/y/m/d/ln/mn', '-g/ge/s/se'
    private static final int NUMBER_OF_OPTION_SIZE_FOR_SECOND_COMMAND = 2;  //second command consists of '-f/l/y/m/d/ln/mn', '-g/ge/s/se'

    private boolean isPrintOn = false;

    private String filterOption = Options.EMPTY_OPTION;
    private String compareOption = Options.EMPTY_OPTION;

    public OptionParser(List<String> options) {
        if (!options.isEmpty()) {
            initializeByNumberOfOptions(options);
        }
    }

    private void initializeByNumberOfOptions(List<String> options) {
        if (options.size() == NUMBER_OF_OPTION_SIZE_FOR_FIRST_COMMAND)
            initFirstCommandOptionParser(options);
        else if (options.size() == NUMBER_OF_OPTION_SIZE_FOR_SECOND_COMMAND)
            initSecondCommandOptionParser(options);
        else
            throw new IllegalArgumentException("Invalid number of options.");
    }

    private void initFirstCommandOptionParser(List<String> options) {
        setPrintOption(options.get(0));
        setFilterOption(options.get(1));
        setCompareOption(options.get(2));
    }

    private void initSecondCommandOptionParser(List<String> options) {
        setFilterOption(options.get(0));
        setCompareOption(options.get(1));
    }

    private void setPrintOption(String commandPrintArgument) {
        if (isNotEmptyOption(commandPrintArgument))
            isPrintOn = true;
    }

    private void setFilterOption(String commandFirstArgument) {
        if (isNotEmptyOption(commandFirstArgument)) {
            filterOption = commandFirstArgument;
        }
    }

    private void setCompareOption(String commandSecondArgument) {
        if (isNotEmptyOption(commandSecondArgument)) {
            compareOption = commandSecondArgument;
        }
    }

    private boolean isNotEmptyOption(String optionValue) {
        return !optionValue.equals(Options.EMPTY_OPTION);
    }

    public boolean isPrintOn() {
        return isPrintOn;
    }

    public String getFilterOption() {
        return filterOption;
    }

    public String getCompareOption() {
        return compareOption;
    }
}
