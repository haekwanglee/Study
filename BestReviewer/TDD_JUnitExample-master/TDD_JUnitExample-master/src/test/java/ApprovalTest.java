import com.spun.util.ObjectUtils;
import com.spun.util.io.FileUtils;
import org.approvaltests.Approvals;
import org.approvaltests.approvers.FileApprover;
import org.approvaltests.namer.ApprovalNamer;
import org.approvaltests.namer.NamedEnvironment;
import org.approvaltests.namer.NamerFactory;
import org.approvaltests.writers.ApprovalTextWriter;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.EnumSource;
import org.junit.jupiter.params.provider.ValueSource;
import org.lambda.functions.Function2;

import java.io.File;
import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class ApprovalTest {
    @Test
    @DisplayName("Test using approvals.verify()")
    void approvedTest(){
        Approvals approvals = new Approvals();
        Person john = new Person("다운", "정");
        Person jane = new Person("하나", "강");

        approvals.verify(john.getFullName() + " "+jane.getFullName());
//        Approvals.verify(john.getFullName() + jane.getFullName());
    }

    @Test
    @DisplayName("Approval Test using temporary file approvals. : FileApprover.apprveTextFile()")
    public void testApproveTextFile()
    {
        File f1 = createFile("file1 text");
        File f2 = createFile("file2 text");
        assertFalse(FileApprover.approveTextFile(f2, f1), "files are different");
        f2 = createFile("file1 text");
        assertTrue(FileApprover.approveTextFile(f2, f1), "files are the same");
    }

    @Test
    @DisplayName("Approval Test using file approvals with non-exsiting file : FileApprover.apprveTextFile()")
    public void testApproveTextFileWithNonExsitantFile()
    {
        File f1 = createFile("1");
        File f2 = new File("no exist");
        assertTrue(f1.exists());
        assertFalse(f2.exists());
        assertFalse(FileApprover.approveTextFile(f2, f1));
    }

    @Test
    @DisplayName("Approval Test using file approvals with exsiting file : FileApprover.apprveTextFile()")
    public void testApproveTextFileWithExsitantFile() throws IOException {
        File f1 = createFile("1");
        File f2 = new File("src\\test\\java\\customApproval.txt");
        if(!f2.exists()){
            f2.createNewFile();
        }
        assertTrue(f1.exists());
        assertTrue(f2.exists());
        assertTrue(FileApprover.approveTextFile(f2, f1));
    }

    private File createFile(String string)
    {
        try
        {
            File f = File.createTempFile("avc", "t");
            FileUtils.writeFile(f, string);
            return f;
        }
        catch (IOException e)
        {
            throw ObjectUtils.throwAsError(e);
        }
    }

    @Test
    @DisplayName("Customize Approval Test : ApprovalTextWriter, ApprovalNamer")
    public void testCustomApprover()
    {
        ApprovalTextWriter writer = new ApprovalTextWriter("Random: ",new String("customApprovedTest"));
        ApprovalNamer namer = Approvals.createApprovalNamer();
        Function2<File, File, Boolean> approveEverything = (r, a) -> true;
        Approvals.verify(new FileApprover(writer, namer, approveEverything));
    }

}

class ParameterizedApprovalTest {

    enum MyEnum {
        FOO, BAR
    }

    @Test
    void testVerifyAllWithEnumSource() {
        Approvals.verifyAll("enum source", MyEnum.values());
    }

    @ParameterizedTest
    @EnumSource(MyEnum.class)
    void testWithEnumSource(MyEnum e) {
        try (NamedEnvironment en = NamerFactory.withParameters(e))
        {
            // your code goes here
            Object output = e;
            Approvals.verify(output);
        }
    }

    @ParameterizedTest
    @ValueSource(strings = {"parameter1", "parameter2", "parameter3"})
    void sampleParameterizedTest(String parameter)
    {
        try (NamedEnvironment en = NamerFactory.withParameters(parameter))
        {
            // your code goes here
            Object output = parameter;
            Approvals.verify(output);
        }
    }
}