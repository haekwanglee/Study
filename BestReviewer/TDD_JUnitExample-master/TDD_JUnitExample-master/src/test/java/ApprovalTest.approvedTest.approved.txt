다운 정 하나 강